src/generate_conf.rs

```diff
@@ -85,9 +85,30 @@ fn generate_config(
         .ok_or_else(|| anyhow!("Invalid NM configuration"))?
         .to_owned();
 
+    populate_connection_ids(&mut interfaces, &config);
+
     Ok((interfaces, config))
 }
 
+fn populate_connection_ids(interfaces: &mut [Interface], config: &NetworkConfig) {
+    for (_, content) in config {
+        let mut config = Ini::new();
+        config
+            .read(content.to_string())
+            .expect("Unable to read nmconnection file");
+        if let Some(interface_name) = config.get("connection", "interface-name") {
+            if let Some(id) = config.get("connection", "id") {
+                interfaces
+                    .iter_mut()
+                    .filter(|x| x.logical_name == interface_name)
+                    .for_each(|interface| {
```

Comment:
The "proper" approach is to use the `find()` method to identify the necessary interface. Also, do you think that we should return an error if an interface is not found here? I think we risk this could silently fail sometimes.